anta/catalog.py

```diff
@@ -335,6 +367,34 @@ def from_list(data: ListAntaTestTuples) -> AntaCatalog:
             raise
         return AntaCatalog(tests)
 
+    def merge(self, catalog: AntaCatalog) -> AntaCatalog:
+        """Merge two AntaCatalog instances.
+
+        Args:
+        ----
+            catalog: AntaCatalog instance to merge to this instance.
+
+        Returns
+        -------
+            A new AntaCatalog instance containing the tests of the two instances.
+        """
+        tests = self.tests
+        tests.extend(catalog.tests)
+        return AntaCatalog(tests=tests)
+
+    def dump(self) -> AntaCatalogFile:
+        """Return an AntaCatalogFile instance from this AntaCatalog instance.
+
+        Returns
+        -------
+            An AntaCatalogFile instance containing tests of this AntaCatalog instance.
+        """
+        root: dict[ImportString[Any], list[AntaTestDefinition]] = {}
+        for test in self.tests:
+            module = test.test.__module__
```

Comment:
Should we add a `module` property in AntaTest to pick this up nicely?

Reply:
I have defined the property in `AntaTest` ABC but this cannot be used before the class is actually instanciated (which is not the case here). I don't have better ideas.